### PR TITLE
ERT Minor Refactor, more ERTs, ERTs can now spawn on subshuttles

### PR DESCRIPTION
**Pull request by meemofcourse**
## About The Pull Request

Ports https://github.com/tgstation/tgstation/pull/71348 and https://github.com/tgstation/tgstation/pull/56345, with code adapted for Shiptest. 

Also adds the following ERTs:
* NT Loss Prevention
* Frontiersmen
* Inteq
* Minutemen
  * Riot Control
  * BARD
  * GOLD Collector
* Syndicate
  * Gorlex Maruders
  * Cybersun Commando
  * Cybersun Paramedic
* SolGov
  * Sonnensoldner
  * Solarian Inspector

## Why It's Good For The Game

adminbus funny

## Changelog

:cl: Zonespace27, Ryll-Ryll, MeeMOfCourse
tweak: Admins can now spawn an unarmed horde of interns in addition to the old armed horde
tweak: Admins can toggle name randomization for ERT's, as well as if they'd like to be spawned in the briefing room at the time of creation
tweak: Admins can now force ERT's to pick the leader role from the most experienced candidates in terms of playtime
tweak: Menus like Create Response Team and Create Portal Storm now toggle boolean options when clicked, rather than open a popup window to select one of the two options
refactor: Centcom Officials are now a full subtype of ERT, rather than being a separate antagonist type with duplicated functionality
admin: ERTs can now have a custom shuttle that they can spawn on, on a toggle while creating said ERT.
admin: Shiptest-Specific ERTs, and respective files.
/:cl:

### CODE REVIEW

File: code/modules/clothing/outfits/ert/frontiersmen_ert.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	gloves = /obj/item/clothing/gloves/combat
+	ears = /obj/item/radio/headset/pirate/alt
+	back = /obj/item/storage/backpack
+	belt = null
+	l_pocket = /obj/item/flashlight/seclite
+	r_pocket = /obj/item/tank/internals/emergency_oxygen/double
+
+	backpack_contents = list(/obj/item/ammo_box/a762=5, /obj/item/grenade/frag=1)
+
+	id_role = "Buccaneer"
```
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/frontiersmen_ert.dm]: Buccaneer is a really pirate brained name, not frontiersman. Give them something more strikingly militaristic.

File: code/modules/clothing/outfits/ert/frontiersmen_ert.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/datum/outfit/centcom/ert/frontiersmen
+	name = "ERT - Frontiersman Basic"
+
+	head = /obj/item/clothing/head/beret/sec/frontier
+	mask = /obj/item/clothing/mask/gas/sechailer/minutemen
+	suit = /obj/item/clothing/suit/armor/vest/bulletproof/frontier
+	suit_store = /obj/item/gun/ballistic/rifle/boltaction
+	uniform = /obj/item/clothing/under/rank/security/officer/frontier
+	shoes = /obj/item/clothing/shoes/combat
+	gloves = /obj/item/clothing/gloves/combat
```
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/frontiersmen_ert.dm]: basic guys don't need insulated gloves

File: code/modules/clothing/outfits/ert/frontiersmen_ert.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	name = "ERT - Frontiersman Engineer"
+
+	back = /obj/item/storage/backpack/industrial
+	belt = /obj/item/storage/belt/utility/full
+	head = /obj/item/clothing/head/hardhat/weldhat
+	suit_store = null
+
+	backpack_contents = list(/obj/item/grenade/c4=3, /obj/item/crowbar/large=1)
+
+	id_role = "Carpenter"
```
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/frontiersmen_ert.dm]: I curse spockye for adding the ember before I finished the Tobermoon. Militarism please.

File: code/modules/clothing/outfits/ert/frontiersmen_ert.dm [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+	uniform = /obj/item/clothing/under/rank/security/officer/frontier/officer
+	head = /obj/item/clothing/head/beret/sec/frontier/officer
+	ears = /obj/item/radio/headset/pirate/alt/captain
+	back = /obj/item/storage/backpack/satchel/leather
+	suit = /obj/item/clothing/suit/armor/frontier
+	suit_store = /obj/item/gun/ballistic/revolver/nagant
+	belt = /obj/item/storage/belt/military/assault
+
+	backpack_contents = list(/obj/item/ammo_box/n762_clip=3, /obj/item/binoculars=1, /obj/item/melee/classic_baton/telescopic)
```
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/frontiersmen_ert.dm]: I do not care for stun combat. Give them like... a nice knife.

File: code/modules/clothing/outfits/ert/inteq_ert.dm
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/inteq_ert.dm:1]: I'd add Inteq to all the id_roles here.

File: code/modules/clothing/outfits/ert/inteq_ert.dm [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/datum/outfit/centcom/ert/inteq
+	name = "ERT - Inteq Rifleman"
+
+	mask = /obj/item/clothing/mask/gas/sechailer/inteq
+	glasses = /obj/item/clothing/glasses/hud/security/sunglasses/inteq
+	uniform = /obj/item/clothing/under/syndicate/inteq
+	suit = /obj/item/clothing/suit/space/hardsuit/syndi/inteq
+	suit_store = /obj/item/gun/ballistic/automatic/assault/ak47/inteq
```
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/inteq_ert.dm:8]: the ak is pretty powerful. It'd be nice to have a semi auto rifle of some flavor that isn't a DMR.

File: code/modules/clothing/outfits/ert/inteq_ert.dm [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+	l_pocket = /obj/item/kitchen/knife/combat
+	r_pocket = /obj/item/flashlight/seclite
+
+	id_role = "Enforcer"
+
+/datum/outfit/centcom/ert/inteq/shotgun
+	name = "ERT - Inteq Shotgunner"
+
+	suit_store = /obj/item/gun/ballistic/shotgun/bulldog/inteq
```
Comment (Erikafox) [inline code/modules/clothing/outfits/ert/inteq_ert.dm]: give them a combat shotgun imo.

**Review by Erikafox — COMMENTED** [review]
assorted just balance thoughts.

File: code/modules/clothing/outfits/ert/inteq_ert.dm
Comment (meemofcourse) [reply code/modules/clothing/outfits/ert/inteq_ert.dm:1]: whar. shouldn't there be distinction between each other

File: code/modules/admin/verbs/one_click_antag.dm [hunk 7]
```diff
@@ -306,79 +315,149 @@
...
-			while(numagents && candidates.len)
-				var/spawnloc = spawnpoints[index+1]
+		// This list will take priority over spawnpoints if not empty
+		var/list/spawn_turfs = list()
+
+		// Takes precedence over spawnpoints[1] if not null
+		var/turf/brief_spawn
+
+		if(!length(candidates))
+			return FALSE
```
Comment (MarkSuckerberg) [inline code/modules/admin/verbs/one_click_antag.dm]:
```suggestion
		if(!length(candidates))
			to_chat(usr, span_warning("No applicants for ERT. Aborting spawn."))
			return FALSE
```

File: code/modules/admin/verbs/one_click_antag.dm [hunk 8]
```diff
@@ -306,79 +315,149 @@
...
+
+		if(ertemplate.spawn_admin)
+			if(isobserver(usr))
+				var/mob/living/carbon/human/admin_officer = new (brief_spawn || spawnpoints[1])
+				var/chosen_outfit = usr.client?.prefs?.brief_outfit
+				usr.client.prefs.copy_to(admin_officer)
+				admin_officer.equipOutfit(chosen_outfit)
+				admin_officer.key = usr.key
+			else
+				to_chat(usr, "<span class='warning'>Could not spawn you in as briefing officer as you are not a ghost!</spawn>")
```
Comment (MarkSuckerberg) [inline code/modules/admin/verbs/one_click_antag.dm]:
```suggestion
				to_chat(usr, span_warning("Could not spawn you in as briefing officer as you are not a ghost!"))
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
just some minor things

File: code/modules/clothing/outfits/ert/nanotrasen_ert.dm [hunk 9]
```diff
@@ -58,13 +60,13 @@
...
 	id = /obj/item/card/id/ert/security
 	suit = /obj/item/clothing/suit/space/hardsuit/ert/sec
-	suit_store = /obj/item/gun/energy/e_gun/hades
 	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
 	gloves = /obj/item/clothing/gloves/tackler/combat/insulated
 	back = /obj/item/storage/backpack/ert/security
 	belt = /obj/item/storage/belt/security/full
 	backpack_contents = list(/obj/item/storage/box/survival/engineer=1,\
 		/obj/item/storage/box/handcuffs=1,\
+		/obj/item/gun/energy/e_gun/stun=1,\
```
Comment (thgvr) [inline code/modules/clothing/outfits/ert/nanotrasen_ert.dm]:
```suggestion
		/obj/item/gun/energy/e_gun/hades=1,\
```

**Review by thgvr — CHANGES_REQUESTED** [review]

**Review by thgvr — APPROVED** [review]

**Comment by Admiral-Gigaton** [comment]
cant wait for an event where the admins just spawn like 8 pills total, load them with syndie/NT ERTs, and send them out to battle to the death

**Comment by thgvr** [comment]
It seems fine in TMs but there's still some issues I believe with erts spawning at centcom. I'll take a thorough look this weekend.

File: code/modules/clothing/outfits/ert/nanotrasen_ert.dm [hunk 10]
```diff
@@ -83,25 +85,27 @@
 	mask = /obj/item/clothing/mask/gas/sechailer/swat
 	backpack_contents = list(/obj/item/storage/box/survival/engineer=1,\
 		/obj/item/storage/box/handcuffs=1,\
-		/obj/item/melee/baton/loaded=1)
-	suit_store = /obj/item/gun/energy/pulse/carbine/loyalpin
+		/obj/item/melee/baton/loaded=1,\
+		/obj/item/gun/energy/pulse/carbine/loyalpin=1)
```
Comment (MarkSuckerberg) [inline code/modules/clothing/outfits/ert/nanotrasen_ert.dm]:
```suggestion
		/obj/item/melee/baton/loaded=1)
		suit_store = /obj/item/gun/energy/pulse/carbine/loyalpin
```
it don't fit

**Review by MarkSuckerberg — APPROVED** [review]
very good pr

**Pull request merged**